### PR TITLE
Implement additional `DualQuaternion` ops and `UnitDualQuaternion`

### PR DESCRIPTION
**Pull request by tpdickso**
This implements `UnitDualQuaternion` as an alternative to `Isometry3` for representing 3D isometries, which also provides the `sclerp` operation which can be used to perform screw-linear interpolation between two unit dual quaternions.

### CODE REVIEW

**Review by chinedufn — COMMENTED** [review]
I did not review everything. Just left one quick question.

File: src/geometry/dual_quaternion.rs
Comment (tpdickso) [reply src/geometry/dual_quaternion.rs:87]: It's idiomatic for `normalize_mut` functions to return the norm of the value prior to normalization, as it's a necessary intermediate step to normalize the value anyway, and it's very often useful alongside the normalized value. For example, in a function where you're performing a raycast, it can be useful to normalize the ray to produce a unit direction vector, while also getting the norm of the ray to scale the time-of-impact factor back to being a parameter along the ray rather than a scaling factor for the direction vector. The [implementation of normalize_mut for vectors](https://docs.rs/nalgebra/0.24.0/nalgebra/base/struct.Matrix.html#method.normalize_mut) works this way too.

It's probably not *as* useful for a dual quaternion, as unit dual quaternions are much more commonly used... but it provides consistency, and I've found a use for it in my project, where I encode similarities instead of isometries by encoding the scale factor in the norm of the real part.

**Comment by chinedufn** [comment]
I'd imagine that @sebcrozet would have an easier time landing this as a bunch of smaller PRs instead of one big one.

I'd be happy to help with reviewing some smaller PRs.

**Review by sebcrozet — APPROVED** [review]
Thank you for this PR!
This a very good-quality PR. It looks like you took the code from the isometry (and/or UnitQuaternion) and transformed it so it fits the `UnitDualQuaternion` type. So overall it is very complete.

I don't have much to request except for a few cosmetic changes (like some variable names in doc-tests). I took the liberty to make these changes myself.

Now I'm waiting for the CI to complete before merging this.

**Pull request merged**
